static/src/javascripts/projects/commercial/modules/hosted/youtube.js

```diff
@@ -0,0 +1,113 @@
+
+define([
+    'commercial/modules/hosted/next-video-autoplay',
+    'common/modules/video/youtube-player',
+    'common/utils/$',
+    'common/utils/detect',
+    'common/utils/mediator',
+    'lodash/collections/contains',
+    'lodash/collections/forEach'
+], function (
+    nextVideoAutoplay,
+    youtubePlayer,
+    $,
+    detect,
+    mediator,
+    contains,
+    forEach
+) {
+    function isDesktop() {
+        return contains(['desktop', 'leftCol', 'wide'], detect.getBreakpoint());
+    }
+
+    function initYoutubeEvents(videoId) {
+        var eventList = ['play', '25', '50', '75', 'end'];
+
+        forEach(eventList, function(event) {
+            mediator.once(event, function() {
+                ophanRecord(event);
+            });
+        });
+
+        function ophanRecord(event) {
+            require(['ophan/ng'], function (ophan) {
+                var eventObject = {
+                    video: {
+                        id: 'gu-video-youtube-' + videoId,
+                        eventType: 'video:content:' + event
+                    }
+                };
+                ophan.record(eventObject);
+            });
+        }
+    }
+
+    function sendPercentageCompleteEvents(youtubePlayer, playerTotalTime) {
+        var quartile = playerTotalTime / 4;
+        var playbackEvents = {
+            'play': 0,
+            '25': quartile,
+            '50': quartile * 2,
+            '75': quartile * 3,
+            'end': playerTotalTime
```

Comment:
No, if we autoplay it manually (by clicking on the overlay). Yes if we will wait until the end of the movie to be auto-redirected automatically. In both situations, we track the autoplay event so it should be easy to detect why the video wasn't watched until the end.